src/PrestaShopBundle/EventListener/FilterCmsPageCategorySearchCriteriaListener.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\EventListener;
+
+use PrestaShop\PrestaShop\Core\Search\Filters\CmsPageCategoryFilters;
+use PrestaShopBundle\Event\FilterSearchCriteriaEvent;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class FilterCmsPageCategorySearchCriteriaListener is responsible for updating CmsCategoryFilters filter with
+ * cms page category id.
+ */
+class FilterCmsPageCategorySearchCriteriaListener
```

Comment:
I might be missing something but why not performing this in the Controller action https://github.com/PrestaShop/PrestaShop/pull/10877/files#diff-dee4848ddf509adc016fd3e905b03f37R91 ?

Splitting the behavior in 2 files make it harder to understand ?

Reply:
Well I might agree with you here - using the event listener is cool way to change default results in this case but it might be unfamiliar for most people where that modified content comes from .
@sarjon  what do you think?